addons/website_blog/static/src/js/blog_tag_add.js

```diff
@@ -0,0 +1,336 @@
+odoo.define('website_blog.blog_post_tag.add', function (require) {
+'use strict';
+
+const {qweb, _t} = require('web.core');
+const options = require('web_editor.snippets.options');
+const {Class: EditorMenuBar} = require('web_editor.editor');
+
+options.registry.TagBlogPostSelect = options.Class.extend({
+    xmlDependencies: (options.Class.prototype.xmlDependencies || []).concat(['/website_blog/static/src/xml/website_blog_tag.xml']),
+    /**
+     * @override
+     */
+    willStart: async function () {
+        const _super = this._super.bind(this);
+        this.blogPostId = this.$target.data('blog-id');
+        this.addList = [];
+        this.unlinkList = [];
+        this.isEditing = false;
+        this.trigger_up('set_blogPost', {
+            blogPostId: this.blogPostId,
+        });
+        await this._getTags();
+        return _super(...arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * @see this.selectClass for params
+     */
+    editList: function (previewMode, widgetValue, params) {
+        this.isEditing = true;
+        this._toggleEditUI();
+        this.rerender = true;
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    saveTagList: function (previewMode, widgetValue, params) {
+        this.isEditing = false;
+        this._saveChanges();
+        this._toggleEditUI();
+        this.rerender = true;
+    },
+    /**
+     * @see this.selectClass for params
+     */
+    confirmNew: function (previewMode, widgetValue, params) {
+        const name = $('[data-name="add_tag_opt"] input').val();
```

Comment:
Obviously not the right way to get a value from the UI... Do it the same way as any other widget: add a method on the input (something like "data-set-created-tag-name"), save the received widgetValue in a variable and use that variable in this `confirmNew` method.